components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -112,4 +180,84 @@ mod tests {
             assert_eq!(output, expect_output, "{:?}", arg);
         }
     }
+
+    #[test]
+    fn test_floor_real() {
+        let cases = vec![
+            (3.0, 3.5),
+            (3.0, 3.7),
+            (3.0, 3.45),
+            (3.0, 3.1),
+            (-4.0, -3.45),
+            (-1.0, -0.1),
+            (std::f64::MAX, std::f64::MAX),
+            (std::f64::MIN, std::f64::MIN),
+        ];
+        for (expected, input) in cases {
+            let arg = Real::from(input);
+            let expected = Real::new(expected).ok();
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(arg)
+                .evaluate::<Real>(ScalarFuncSig::FloorReal)
+                .unwrap();
+            assert_eq!(expected, output);
+        }
+    }
+
+    #[test]
+    fn test_floor_dec_to_dec() {
+        let cases = vec![
+            ("9223372036854775808", "9223372036854775808"),
+            ("123", "123.456"),
+            ("-124", "-123.456"),
+        ];
+
+        for (expected, input) in cases {
+            let arg = input.parse::<Decimal>().ok();
+            let expected = expected.parse::<Decimal>().ok();
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(arg)
+                .evaluate::<Decimal>(ScalarFuncSig::FloorDecToDec)
+                .unwrap();
+            assert_eq!(expected, output);
+        }
+    }
+
+    #[test]
+    fn test_floor_dec_to_int() {
+        let cases = vec![
+            (123, "123.456"),
+            (1, "1.23"),
+            (-2, "-1.23"),
+            (std::i64::MIN, "-9223372036854775808"),
+        ];
+        for (expected, input) in cases {
+            let arg = input.parse::<Decimal>().ok();
+            let expected = Some(expected);
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(arg)
+                .evaluate::<Int>(ScalarFuncSig::FloorDecToInt)
+                .unwrap();
+            assert_eq!(expected, output);
+        }
+    }
+
+    #[test]
+    fn test_floor_int_to_int() {
+        let cases = vec![
+            (1, 1),
```

Comment:
What about the numbers less than 0?

Reply:
Thanks, I will add this case.

Reply:
Wait! I have already add `std::i64::MIN` to test cases.